stock_move_line_serial_unique/tests/test_stock_move_line_serial_unique.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import ValidationError
+from odoo.tests.common import TransactionCase
+
+
+class TestStockQuantSerialUnique(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        company = cls.env.ref("base.main_company")
+        cls.stock_location = cls.env.ref("stock.stock_location_stock")
+        cls.vendor_location = cls.env.ref("stock.stock_location_suppliers")
+        cls.product1 = cls.env["product.product"].create(
+            {"name": "Product A", "type": "product", "tracking": "serial"}
+        )
+        lot1 = cls.env["stock.lot"].create(
+            {"company_id": company.id, "product_id": cls.product1.id, "name": "Lot1"}
+        )
+        cls.env["stock.quant"].create(
+            {
+                "product_id": cls.product1.id,
+                "location_id": cls.stock_location.id,
+                "lot_id": lot1.id,
+                "quantity": 1,
+            }
+        )
+        cls.owner1 = cls.env["res.partner"].create({"name": "Test Company"})
+
+    def _create_moveline(
```

Comment:
Split this method into _create_picking( ) and _create_moveline(), or better yet, create pickings upfront in the setup for slightly better performance.